From 4b667d3b0a9cfa408ed0975e85ad279db543e9c5 Mon Sep 17 00:00:00 2001
From: Douglas Wilson <douglas.wilson@gmail.com>
Date: Thu, 13 Jul 2017 12:09:41 +1200
Subject: [PATCH] Remove imports of Data.Typeable.Internal

This module is removed from base in ghc-8.2. Luckily, it doesn't appear to have
been strictly necessary.
---
 src/Data/Store/Internal.hs |  2 +-
 src/Data/Store/Version.hs  | 10 +++-------
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/src/Data/Store/Internal.hs b/src/Data/Store/Internal.hs
index aa05625..7cee122 100644
--- a/src/Data/Store/Internal.hs
+++ b/src/Data/Store/Internal.hs
@@ -108,7 +108,7 @@ import qualified Data.Text.Array as TA
 import qualified Data.Text.Foreign as T
 import qualified Data.Text.Internal as T
 import qualified Data.Time as Time
-import           Data.Typeable.Internal (Typeable)
+import           Data.Typeable (Typeable)
 import qualified Data.Vector as V
 import qualified Data.Vector.Mutable as MV
 import qualified Data.Vector.Storable as SV
diff --git a/src/Data/Store/Version.hs b/src/Data/Store/Version.hs
index 575a7ac..e378978 100644
--- a/src/Data/Store/Version.hs
+++ b/src/Data/Store/Version.hs
@@ -44,7 +44,6 @@ import qualified Data.Text as T
 import           Data.Text.Encoding (encodeUtf8, decodeUtf8, decodeUtf8With)
 import           Data.Text.Encoding.Error (lenientDecode)
 import qualified Data.Text.IO as T
-import           Data.Typeable.Internal (TypeRep(..))
 import           Data.Word (Word32)
 import           GHC.Generics (Generic)
 import           Language.Haskell.TH
@@ -230,12 +229,9 @@ getStructureInfo' ignore renames _ = do
                 return (error "unexpected evaluation")
 
 showsQualTypeRep :: M.Map String String -> Int -> TypeRep -> ShowS
-#if MIN_VERSION_base(4,8,0)
-showsQualTypeRep renames p (TypeRep _ tycon _ tys) =
-#else
-showsQualTypeRep renames p (TypeRep _ tycon tys) =
-#endif
-    case tys of
+showsQualTypeRep renames p tyrep =
+  let (tycon, tys) = splitTyConApp tyrep
+  in case tys of
         [] -> showsQualTyCon renames tycon
         [x] | tycon == tcList -> showChar '[' . showsQualTypeRep renames 0 x . showChar ']'
           where
